### DISCUSSION TITLE
Allow RegistryStrategy to be overwritten by datasource config

### DISCUSSION THREAD

**Discussion by rmacconn** [renovatebot/renovate#22743] [Suggest an Idea]
### Tell us more.

Currently each datasource overrides the registryStrategy but there is no way for users to set the strategy from the Renovate configuration.  We would like the ability to set the registryStrategy for a datasource from the configuration file.

```
  "packageRules": [
    {
      "matchDatasources": ["maven"],
      "registryStrategy": "hunt",
      "registryUrls": [
        "https://registry1/",
        "https://registry2/",
      ]
    },

```

**Current Impact:**
We have several third-party registries which provide libraries our products depend on but these registries also mirror public registries.  Since Renovate sets the registryStrategy for Maven to 'merge', we end up fetching 60GB of packages through these third party registries when Renovate runs since all registries get searched.  This does not happen when maven runs during the regular build process.

We would like to be able to override the registryStrategy to 'hunt' so that we only fetch these packages from our primary public registry which is set as the first value in our registryUrls list.

**Comment by a1flecke** [comment]
This would be a big help.  Since Maven [fetches in repositories in a particular order and then stops when a result is found](https://maven.apache.org/guides/mini/guide-multiple-repositories.html#repository-order) then I would expect Renovate to as well.

**Comment by rarkins** [comment]
Created #22744 to fix Maven registryStrategy
